lib/mini_racer.rb

```diff
@@ -1,5 +1,12 @@
 require "mini_racer/version"
-require "mini_racer_extension"
+require "mini_racer_loader"
+
+ext_filename = "mini_racer_extension.#{RbConfig::CONFIG['DLEXT']}"
+ext_path = $LOAD_PATH.map { |p| (Pathname.new(p) + ext_filename) }.find { |p| p.file? }
```

Comment:
do we need to walk the whole load path here? can we just rely on __FILE___ or __dir__ here?

Reply:
This would point us to `lib`, or possibly `ext/mini_racer_extension` indeed, but in a hardcoded way. Ruby has this funny way of using one or more of those paths:

```
$ find ~/.gem/ruby/2.7.1 -name 'mini_racer_extension.bundle' | xargs ls -l 
-rwxr-xr-x  1 lloeki  staff  31947216 Sep 30 12:19 /Users/lloeki/.gem/ruby/2.7.1/extensions/x86_64-darwin-19/2.7.0-static/mini_racer-0.3.1/mini_racer_extension.bundle
-rwxr-xr-x  1 lloeki  staff  31947216 Sep 30 12:19 /Users/lloeki/.gem/ruby/2.7.1/gems/mini_racer-0.3.1/ext/mini_racer_extension/mini_racer_extension.bundle
-rwxr-xr-x  1 lloeki  staff  31947216 Sep 30 12:19 /Users/lloeki/.gem/ruby/2.7.1/gems/mini_racer-0.3.1/lib/mini_racer_extension.bundle
```

To reduce the path set down from `$LOAD_PATH`, we could use `Gem::Specification#require_paths` instead:

```
$ bundle exec ruby -rmini_racer -e 'p Gem.loaded_specs["mini_racer"].require_paths'                                                
ruby -rmini_racer -e 'p Gem.loaded_specs["mini_racer"].require_paths'       
["/Users/lloeki/.gem/ruby/2.7.1/extensions/x86_64-darwin-19/2.7.0-static/mini_racer-0.3.1", "lib", "ext"]
```

That'd make the marked line become:

```
Gem.loaded_specs["mini_racer"].require_paths.map { |p| (Pathname.new(p) + ext_filename) }.find { |p| p.file? }
```

WDYT?

Reply:
I have implemented the change.

Reply:
that is much better! thanks.